hmcts/pre-api
# S28 3352: Recording Duration From `.mpd` File

### JIRA ticket(s)
- S28-3352


### Change description
- Add `RecordingListener` which automatically updates recording's duration on recording creation event. Sources of this event: 
    - Creation of new recording via `PUT /recordings/{id}`
    - Creation of new recording via `PUT /media-service/live-event/end/{id}` (when `RECORDING_AVAILABLE`)
    - Creation of new recording via `PUT /media-service/generate-asset` (edit process)
- Get recording duration from the `.mpd` file generated for an asset alongside the `.mp4`. 
    - `.mpd` file contains XML. Recording duration is under the attribute `mediaPresentationDuration`. The value for this is stored in ISO 8601 duration format.
    - If no container or not `.mpd` file found, then duration is set to `null`.
    - If error occurs when getting attribute from `.mpd` file , then duration is set to `null` with a error logged. 




> [!IMPORTANT]
> This PR requires manual testing by QA. Please notify the QA team @hmcts/pre-rec-evidence-qa.

**QA instructions**
- Create a recording via `PUT /recordings/{id}` where the `id` matches the id of a recording that has a container in storage (set duration to null). See created successfully (201 Created). Call `GET /recordings/{id}`, get 200 response. See value for recording's duration is not null and matches the duration of the recording (to double check this value either download the `.mp4` from the container and look at the playtime OR play asset in app + see duration).
- Create a recording via `PUT /recordings/{id}` where the `id` matches the id of a recording that has a container in storage (set duration to non-null value). See created successfully. Call `GET /recordings/{id}`, get 200 response. See value for recording's duration is not null but matches the value you set INSTEAD OF VALUE OF DURATION AS FOUND IN MPD.
- Create a recording via  `PUT /recordings/{id}` with an `id` that does not match an `id` of a valid recording (id not a container in final storage). See created successfully. Call `GET /recordings/{id}`, get 200 response. See value for recording's duration is null
- Start a recording in MK. Do CVP recording. End live event. Call `GET /recordings/{id}` for the created recording. See duration is not null + duration is correct.
- Create an edit of a recording. After calling generate asset endpoint, call the `GET /recordings/{id}` endpoint and see duration is not null + duration is correct.
